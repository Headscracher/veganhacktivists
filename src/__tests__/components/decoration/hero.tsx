import React from 'react';
import { render } from '@testing-library/react';
import Hero from '../../../components/decoration/hero';

it('renders correctly when proving a URL string', () => {
  const { asFragment } = render(
    <Hero
      imageBackground="https://via.placeholder.com/500x500"
      alignment="right"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('renders correctly when proving static image data', () => {
  const { asFragment } = render(
    <Hero
      imageBackground={{
        src: 'https://via.placeholder.com/500x500',
        width: 500,
        height: 500,
      }}
      alignment="left"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
