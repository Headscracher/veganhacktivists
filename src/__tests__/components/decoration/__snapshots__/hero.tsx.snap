// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when providing a URL string 1`] = `
<DocumentFragment>
  <div
    class="relative md:-top-20 md:-mb-20 p-10 md:p-32 bg-scroll bg-cover bg-no-repeat flex justify-end"
    style="background-image: url(https://via.placeholder.com/500x500);"
  >
    <div
      class="flex flex-col justify-center w-1/2"
    />
  </div>
</DocumentFragment>
`;

exports[`should render correctly when providing static image data 1`] = `
<DocumentFragment>
  <div
    class="relative md:-top-20 md:-mb-20 p-10 md:p-32 bg-scroll bg-cover bg-no-repeat flex justify-start"
    style="background-image: url(https://via.placeholder.com/500x500); width: 500px; height: 500px;"
  >
    <div
      class="flex flex-col justify-center w-1/2"
    />
  </div>
</DocumentFragment>
`;
